src/main/java/org/c02e/jpgpj/Encryptor.java

```diff
@@ -408,13 +411,68 @@ public void encrypt(InputStream plaintext, OutputStream ciphertext)
      * @throws PassphraseException if an incorrect passphrase was supplied
      * for one of the signing keys.
      */
-    public void encrypt(InputStream plaintext, OutputStream ciphertext,
-    FileMetadata meta) throws IOException, PGPException {
+    public void encrypt(
+        InputStream plaintext, OutputStream ciphertext, FileMetadata meta)
+            throws IOException, PGPException {
+        if (meta == null) {
+            meta = new FileMetadata();
+        }
+
+        try (OutputStream targetStream = prepareCiphertextOutputStream(ciphertext, meta)) {
+            // copy plaintext bytes into encryption pipeline
+            copy(plaintext, targetStream, null /* signer is inside the target */, meta);
+        }
+    }
+
+    public OutputStream prepareCiphertextOutputStream(File ciphertext)
+            throws IOException, PGPException {
+        // delete old output file
+        if (ciphertext.delete()) {
+            log.debug("prepareCiphertextOutputStream - deleted {}", ciphertext);
+        }
+
+        FileMetadata meta = new FileMetadata(ciphertext);
+        OutputStream fileStream = null;
+        try {
+            fileStream = new FileOutputStream(ciphertext);
+            OutputStream wrapper = prepareCiphertextOutputStream(fileStream, meta);
+            fileStream = null;  // avoid auto-close at finally clause
+            return wrapper;
+        } catch(Exception e) {
+            // delete output file if anything went wrong
+            if (fileStream != null) {
+                ciphertext.delete();
+            }
+            throw e;
+        } finally {
+            if (fileStream != null) {
+                fileStream.close();
+            }
+        }
+    }
+
+    /**
+     * Prepares an {@link OutputStream} - so that everything written to it
+     * is encrypted+compressed+signed - according to the encryptor's configuration
```

Comment:
For this method's javadocs, I'm thinking it would help to have some more free-form descriptive text about the distinction between the original output stream passed to the method and the wrapper output stream returned from it -- in particular to emphasize that users need to close both the wrapper stream and the original stream. How about something like this?:

> Builds a new wrapper {@link OutputStream} to wrap the original specified {@link OutputStream}, where everything written to the wrapper stream is encrypted+compressed+signed according to the encryptor's configuration, and then written to the original stream. Closing the wrapper stream finalizes the encryption and signature, and finishes writing all the wrapper stream's content to the original stream, but does not close the original stream; the original stream must be closed separately.

Reply:
Good idea